woocommerce/woocommerce
# Add dismiss button to coming soon banner

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This is a follow-up PR for https://github.com/woocommerce/woocommerce/pull/46096 to add dismiss button.

- Added a new REST API endpoint to have a dismiss endpoint
- Display the coming soon banner only on the store pages when `Restrict to store pages only` option is checked.




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Preparation

1. Create a new JN site with `launch-your-store` feature flag enabled.
2. Go to `WooCommerce -> Settings` and make sure `Coming soon` is selected.
3. Go to `WooCommerce -> Settings` and make sure `Restrict to store pages only` is selected.
4. Visit your site's home and confirm the coming soon banner is not rendered.
5. Visit your site's shop page (/shop) and confirm the coming soon banner is rendered.
6. Disable `Restrict to store pages only` option.
7. Visit your site's home and confirm the coming soon banner is rendered.
8. Click `X` button on the coming soon page.
9. Refresh the page and confirm the banner is no longer rendered.
10. Confirm `woocommerce_coming_soon_banner_dismissed` option is set to `yes`





### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Implement dismiss button for the coming soon banner and restrict the banner on the store pages only when 'Restrict to store pages only' option is selected.

#### Comment 

</details>
